prometheus-to-sd/main.go

```diff
@@ -150,13 +151,30 @@ func main() {
 		options = append(options, option.WithTokenSource(ts))
 	}
 
+	uDomain := ""
 	if *apioverride != "" {
+		glog.Infof("Stackdriver API endpoint is overridden to %s", *apioverride)
+
 		// option.WithEndpoint() only works with "host:port"
 		endpoint, err := getEndpoint(*apioverride)
 		if err != nil {
 			glog.Fatalf("Error parsing --api-override: %s, with error: %v", *apioverride, err)
 		}
 		options = append(options, option.WithEndpoint(endpoint))
+
+		// infer universe domain from Stackdriver api endpoint
+		_, domain, found := strings.Cut(*apioverride, ".")
+		if found {
+			uDomain = strings.TrimSuffix(domain, "/")
+		}
+	}
+	// override inferred universe domain with specified value
+	if *universeDomain != "" {
+		uDomain = *universeDomain
+	}
+	if uDomain != "" {
+		glog.Infof("Universe domain is %s", uDomain)
+		options = append(options, option.WithUniverseDomain(uDomain))
```

Comment:
nit: only add WithUniverseDomain when universeDomain is not ""

Reply:
uDomain is an variable that can be overridden by the command line flag universeDomain.

Reply:
Why does it need to add the option for the default universe?

Reply:
It's default universe when it's not specified in the client options.

Reply:
Why was it working without this option?

Reply:
Discussed offline, agreed on the approach.

Reply:
Thank you for the prompt review.